app/Observers/CampaignObserver.php

```diff
@@ -159,8 +160,10 @@ public function saved(Campaign $campaign)
      */
     public function deleted(Campaign $campaign)
     {
-        SearchCleanupService::cleanup($campaign);
-        ImageService::cleanup($campaign);
+        if ($campaign->isForceDeleting()) {
```

Comment:
Unless you're registering the Observer in the command, this will never get called

Reply:
The observers do get triggered, for example, campaign members are deleted from db when permanently deleting, and that's also handled with the observer.